client/src/Component/MainTable.js

```diff
@@ -73,47 +86,109 @@ const MainTable = (props) => {
 						return (
-							<tr key={client.client_id}>
-								<td>
-									<Checkbox
-										isChecked={props.selectedClients.includes(client.client_id)}
-										label={client.name}
-										onChange={(checked) => props.onHandleSelectedUserState(checked, client.client_id)}
-									/>
-								</td>
-								<td>{client.client_id}</td>
-								<td>{new Date(client.date_added).toLocaleDateString()}</td>
-								<td className="fw-bold">{client.name}</td>
-								<td>{client.phone_number}</td>
-								<td className="align-bikes-needed">{client.bikes_needed}</td>
-								<td className="fs-5"><span className={bookingStatusStyle[client.booking_status]}>{bookingStatus[client.booking_status]}</span></td>
-								<td>{validDate(client.pick_up_date)}</td>
-								<td>{residencyStatus[client.residency_status]}</td>
-								<td>{client.country_of_origin}</td>
-								<td>{client.time_in_scotland}</td>
-								<td>{client.language_spoken}</td>
-								<td>{speakEnglish(client.english_speaker)}</td>
-								<td>{client.english_skill_level}</td>
-								<td>{client.gender}</td>
-								<td>{validDate(client.date_of_birth)}</td>
-								<td>{client.postcode}</td>
-								<td>{client.referring_agency}</td>
-							</tr>
+							<Fragment key={client.client_id}>
+								<tr>
+									<td>
+										<Checkbox
+											isChecked={props.selectedClients.includes(
+												client.client_id
+											)}
+											label={client.name}
+											onChange={(checked) =>
+												props.onHandleSelectedUserState(
+													checked,
+													client.client_id
+												)
+											}
+										/>
+									</td>
+									<td>{client.client_id}</td>
+									<td>{new Date(client.date_added).toLocaleDateString()}</td>
+									<td className="fw-bold">
+										<button
+											className="border-0 bg-transparent me-2"
+											onClick={() => {
+												handleRowClick(i);
+											}}
+										>
+											<svg
+												xmlns="http://www.w3.org/2000/svg"
+												width="16"
+												height="16"
+											>
+												{expandedRowIndex === i
+													? arrowPositionDown
+													: arrowPositionUp}
+											</svg>
+											<span className="visually-hidden">
+												{expandedRowIndex === i
+													? "Show less details"
+													: "Show more details"}
+											</span>
+										</button>
+										<span>{client.name}</span>
+									</td>
+									<td>{client.phone_number}</td>
+									<td className="align-bikes-needed">{client.bikes_needed}</td>
+									<td className="fs-5">
+										<span className={bookingStatusStyle[client.booking_status]}>
+											{bookingStatus[client.booking_status]}
+										</span>
+									</td>
+									<td>{validDate(client.pick_up_date)}</td>
+								</tr>
+								<tr hidden={expandedRowIndex !== i}>
+									<td colSpan="9">
+										<listGroup as="ol">
+											<ListGroup.Item action variant="light">
```

Comment:
```suggestion
											<ListGroup.Item as="li" variant="light">
```
- `action` is used if we want the list item to be a button or link, but in this case we don't want anything to happen so it doesn't make sense to use `action`
- We want to render the list items as `li` elements so we can do `as="li"`. Without `as="li"` it looks like it's rendered as `div` elements

These apply to all the other `ListGroup.Item` elements

Reply:
Thank you Coung I never thought in that way, thank you for make me realise the main purpose of each element.